File: exo8/exo8.py
class exo8:
    @staticmethod
    def triangle(hauteur, char):
        for i in range(0,hauteur):
            for j in range(0, i+1):
                print char[0],
            print
    
    @staticmethod       
    def main():
        h = 5   
        try:
            h = int(input("Veuillez entrer la hauteur du triangle : "))
        except ValueError:
            print("\tTaille invalide, utilisation de la valeur par defaut : 5")

        c = raw_input("Veuillez entrer le caractere a afficher : ")
        
        exo8.triangle(h,c)
        
exo8.main()

File: exo8/exo8.php
<?php

final class exo8 {
    private static function triangle($hauteur = 5, $char = 'm') {
        for ($i = 0; $i < $hauteur; $i++) {
            for ($j = 0; $j <= $i; $j++) {
                echo $char[0] . " ";
            }
            echo "\n";
        }
    }
    
    public static function main() {
        
        echo "Veuillez entrer la hauteur du triangle : ";
        $h = readline();
        if (!is_numeric($h)) {
            echo "\tTaille invalide, utilisation de la valeur par défaut : 5\n";
            $h = 5;
        }
        
        echo "Veuillez entrer le caractère à afficher : ";
        $c = readline();
        
        self::triangle($h, $c);
    }
}

exo8::main();